src/rwr.py

```diff
@@ -0,0 +1,155 @@
+        edges.to_csv(filename_map['edges'], sep=" ", index=False, columns=["Interactor1","Interactor2","Weight"])
+
+
+    # Skips parameter validation step
+    @staticmethod
+    def run(edges=None, sources=None, targets = None, output_file = None, df : float = 0.85, f : str = 'min' , threshold : float = 0.0001, singularity=False):
+        """
+        Run RandomWalk with Docker
+        @param nodetypes:  input node types with sources and targets (required)
+        @param network:  input network file (required)
+        @param output_file: path to the output pathway file (required)
+        @param df: damping factor for restarting (default 0.85) (optional)
+        @param f: selection function (default 'min') (optional)
+        @param threshold: threshold for constructing the final pathway (default 0.0001) (optional)
+        @param singularity: if True, run using the Singularity container instead of the Docker container
+        """
+
+        if not edges or not sources or not targets or not output_file:
+            raise ValueError('Required RandomWalk arguments are missing')
+
+        work_dir = '/spras'
+
+        # Each volume is a tuple (src, dest) - data generated by Docker
+        volumes = list()
+
+        bind_path, edges_file = prepare_volume(edges, work_dir)
+        volumes.append(bind_path)
+
+        bind_path, sources_file = prepare_volume(sources, work_dir)
+        volumes.append(bind_path)
+
+        bind_path, targets_file = prepare_volume(targets, work_dir)
+        volumes.append(bind_path)
+
+
+        out_dir = Path(output_file).parent
+        # RandomWalk requires that the output directory exist
+        out_dir.mkdir(parents=True, exist_ok=True)
+        bind_path, mapped_out_dir = prepare_volume(str(out_dir), work_dir)
+        volumes.append(bind_path)
+        mapped_out_prefix= mapped_out_dir + '/out'  # Use posix path inside the container
+
+
+        command = ['python',
+                   '/RandomWalk/random_walk.py',
+                   '--edges_file', edges_file,
+                   '--sources_file', sources_file,
+                   '--targets_file', targets_file,
+                   '--damping_factor', str(df),
+                   '--selection_function', f,
+                   '--threshold', str(threshold),
+                   '--output_file', mapped_out_prefix]
+
+        print('Running RandomWalk with arguments: {}'.format(' '.join(command)), flush=True)
+
+
+        container_framework = 'singularity' if singularity else 'docker'
+        out = run_container(container_framework,
+                            'erikliu24/rwwr',
+                            command,
+                            volumes,
+                            work_dir)
+        print(out)
+
+        output = Path(out_dir, 'out')
+        output.rename(output_file)
+
+        # From edge_output_file, construct a pathway file in the universal format
```

Comment:
do these comments need to be here?

Reply:
It was simply a spontaneous thought regarding the construction of a pathway. Please disregard it as I intend to delete it.